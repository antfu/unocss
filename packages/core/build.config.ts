import { defineBuildConfig } from 'unbuild'

export default defineBuildConfig({
  entries: [
    'src/index',
  ],
  clean: true,
  declaration: true,
  externals: [
    'unconfig',
    'magic-string-extra',
  ],
  rollup: {
    emitCJS: true,
  },
})
