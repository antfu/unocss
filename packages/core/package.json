{
  "name": "@unocss/core",
  "version": "0.26.4",
  "description": "The instant on-demand Atomic CSS engine.",
  "keywords": [
    "unocss",
    "atomic-css",
    "atomic-css-engine",
    "css",
    "tailwind",
    "windicss"
  ],
  "homepage": "https://github.com/unocss/unocss/tree/main/packages/core#readme",
  "bugs": {
    "url": "https://github.com/unocss/unocss/issues"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/unocss/unocss.git",
    "directory": "packages/core"
  },
  "funding": "https://github.com/sponsors/antfu",
  "author": "Anthony Fu <anthonyfu117@hotmail.com>",
  "sideEffects": false,
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "require": "./dist/index.cjs",
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.ts"
    }
  },
  "main": "dist/index.cjs",
  "module": "dist/index.mjs",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "unbuild",
    "stub": "unbuild --stub"
  },
  "devDependencies": {
    "magic-string": "^0.25.7",
    "unconfig": "^0.3.1"
  }
}
