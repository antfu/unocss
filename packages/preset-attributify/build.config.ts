import { defineBuildConfig } from 'unbuild'

export default defineBuildConfig({
  entries: [
    'src/index',
    'src/variants',
  ],
  clean: true,
  declaration: true,
})
