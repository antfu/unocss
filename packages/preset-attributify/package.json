{
  "name": "@unocss/preset-attributify",
  "version": "0.27.0",
  "description": "Attributify preset for UnoCSS",
  "keywords": [
    "unocss",
    "unocss-preset"
  ],
  "homepage": "https://github.com/unocss/unocss/tree/main/packages/preset-attributify#readme",
  "bugs": {
    "url": "https://github.com/unocss/unocss/issues"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/unocss/unocss.git",
    "directory": "packages/preset-attributify"
  },
  "funding": "https://github.com/sponsors/antfu",
  "author": "Anthony Fu <anthonyfu117@hotmail.com>",
  "sideEffects": false,
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "require": "./dist/index.cjs",
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.ts"
    }
  },
  "main": "dist/index.cjs",
  "module": "dist/index.mjs",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "unbuild",
    "stub": "unbuild --stub"
  },
  "dependencies": {
    "@unocss/core": "workspace:*"
  }
}
