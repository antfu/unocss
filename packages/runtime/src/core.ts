import { initUnocssRuntime } from './factory'

initUnocssRuntime()
