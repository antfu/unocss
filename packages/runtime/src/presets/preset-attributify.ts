import type { AttributifyOptions } from '@unocss/preset-attributify'
import presetAttributify from '@unocss/preset-attributify'

window.__unocss_runtime = window.__unocss_runtime ?? { presets: {} }
window.__unocss_runtime.presets = Object.assign(window.__unocss_runtime?.presets ?? {}, {
  presetAttributify: (options: AttributifyOptions) => presetAttributify(options),
})
