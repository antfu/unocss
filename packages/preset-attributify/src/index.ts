import type { Preset } from '@unocss/core'
import { extractorSplit } from '@unocss/core'
import { extractorAttributify } from './extractor'
import type { AttributifyOptions } from './types'
import { variantAttributify, variantPseudoClasses } from './variants'

export * from './extractor'
export * from './variants'
export * from './types'

const preset = (options?: AttributifyOptions): Preset => {
  const variants = [
    variantAttributify(options),
    variantPseudoClasses,
  ]
  const extractors = [
    extractorAttributify(options),
  ]

  if (!options?.strict)
    extractors.unshift(extractorSplit)

  return {
    name: '@unocss/preset-attributify',
    variants,
    extractors,
    options,
  }
}

export default preset
