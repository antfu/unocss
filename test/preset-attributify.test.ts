import { createGenerator, toEscapedSelector as e } from '@unocss/core'
import presetUno from '@unocss/preset-uno'
import { autocompleteExtractorAttributify, presetAttributify, variantAttributify } from '@unocss/preset-attributify'
import { describe, expect, test } from 'vitest'

describe('attributify', async () => {
  const uno = createGenerator({
    presets: [
      presetAttributify({ strict: true }),
      presetUno({ attributifyPseudo: true }),
    ],
    rules: [
      [/^custom-(\d+)$/, ([_, value], { rawSelector }) => {
        // return a string instead of an object
        const selector = e(rawSelector)
        return `
  ${selector} {
    font-size: ${value}px;
  }
  /* you can have multiple rules */
  ${selector}::after {
    content: 'after';
  }
  .foo > ${selector} {
    color: red;
  }
  /* or media queries */
  @media (min-width: 680px) {
    ${selector} {
      font-size: 16px;
    }
  }
  `
      }],
    ],
  })

  describe('cases', () => {
    const cases = import.meta.glob('./cases/preset-attributify/*/input.html', { as: 'raw' })
    for (const [path, input] of Object.entries(cases)) {
      test(path, async () => {
        const { css } = await uno.generate(await input(), { preflights: false })
        await expect(css).toMatchFileSnapshot(path.replace('input.html', 'output.css'))
      })
    }
  })

  const fixture1 = (await import('./cases/preset-attributify/case-1/input.html?raw')).default

  test('variant', async () => {
    const variant = variantAttributify({
      prefix: 'un-',
      prefixedOnly: false,
      strict: true,
    })

    const promises = Array.from(await uno.applyExtractors(fixture1) || [])
      .map(async (i) => {
        const r = await variant.match(i, {} as any)
        return typeof r === 'string' ? r : r ? r.matcher : r
      })

    expect(await Promise.all(promises))
      .toMatchSnapshot()
  })

  test('prefixedOnly', async () => {
    const uno = createGenerator({
      presets: [
        presetAttributify({ strict: true, prefix: 'un-', prefixedOnly: true }),
        presetUno({ attributifyPseudo: true }),
      ],
    })

    const { css } = await uno.generate(fixture1, { preflights: false })
    expect(css).toMatchSnapshot()
  })

  test('autocomplete extractor', async () => {
    const res = await autocompleteExtractorAttributify.extract({
      content: fixture1,
      cursor: 187,
    })

    expect(res).not.toBeNull()

    expect(res!.extracted).toMatchInlineSnapshot('"bg-blue-400"')
    expect(res!.transformSuggestions!([`${res!.extracted}1`, `${res!.extracted}2`]))
      .toMatchInlineSnapshot(`
        [
          "blue-4001",
          "blue-4002",
        ]
      `)

    const reversed = res!.resolveReplacement(`${res!.extracted}1`)
    expect(reversed).toMatchInlineSnapshot(`
      {
        "end": 192,
        "replacement": "blue-4001",
        "start": 184,
      }
    `)

    expect(fixture1.slice(reversed.start, reversed.end))
      .toMatchInlineSnapshot('"blue-400"')
  })

  test('with trueToNonValued', async () => {
    const uno = createGenerator({
      presets: [
        presetAttributify({ trueToNonValued: true }),
        presetUno(),
      ],
    })
    const { css } = await uno.generate(`
      <div flex="true"></div>
      <div grid="~ cols-2"></div>
    `, { preflights: false })
    expect(css).toMatchSnapshot()
  })

  test('with incomplete element', async () => {
    await uno.generate('<div class="w-fullllllllllllll"')
  }, 20)

  test('merge attribute name and value-only', async () => {
    const { css } = await uno.generate(`
      <div bg="[&:nth-child(3)]:[#123456]"></div>
      <div class="foo" bg="[&.foo]:[&:nth-child(3)]:[#123]"></div>
    `, { preflights: false })
    expect(css).toMatchSnapshot()
  })
})
