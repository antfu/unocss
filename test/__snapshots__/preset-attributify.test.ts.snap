// Vitest Snapshot v1

exports[`attributify > autocomplete extractor 5`] = `
"
<button
  layer-base=\\"c-white/10 hover:c-black/20\\"
  sm=\\"[color:red]\\"
  md=\\"[--var:var(--another)]\\"
  lg=\\"bg-blue-600\\"
  class=\\"absolute fixed\\"
  important=\\"text-red bg-red\\"
  bg=\\"blue-4001 hover:blue-500 dark:!blue-500 dark:hover:blue-600\\"
  text=\\"sm white\\"
  !leading-4
  flex=\\"!~ col\\"
  p=\\"t-2\\"
  pt=\\"2\\"
  border=\\"rounded-xl x-1 x-style-dashed\\"
  :font=\\"foo > bar ? 'mono' : 'sans'\\"
  v-bind:p=\\"y-2 x-4\\"
  border=\\"2 rounded blue-200\\"
  un-children=\\"m-auto\\"
  pt2 rounded-sm
  inline-block
  transform
  translate-x-100%
  translate-y-[10%]
  rotate-30
  after=\\"content-unocss\\"
  rotate-60=\\"\\" ma=\\"\\"
  m='\`
  1 2
  3
\`'
>
  Button
</button>
"
`;

exports[`attributify > compatible with full controlled rules 1`] = `
"/* layer: default */

  [custom~=\\"\\\\31 \\"] {
    font-size: 1px;
  }
  /* you can have multiple rules */
  [custom~=\\"\\\\31 \\"]::after {
    content: 'after';
  }
  .foo > [custom~=\\"\\\\31 \\"] {
    color: red;
  }
  /* or media queries */
  @media (min-width: 680px) {
    [custom~=\\"\\\\31 \\"] {
      font-size: 16px;
    }
  }
  

  .custom-2 {
    font-size: 2px;
  }
  /* you can have multiple rules */
  .custom-2::after {
    content: 'after';
  }
  .foo > .custom-2 {
    color: red;
  }
  /* or media queries */
  @media (min-width: 680px) {
    .custom-2 {
      font-size: 16px;
    }
  }
  "
`;

exports[`attributify > extractor1 1`] = `
Set {
  "[layer-base~=\\"c-white/10\\"]",
  "[layer-base~=\\"hover:c-black/20\\"]",
  "[sm~=\\"[color:red]\\"]",
  "[md~=\\"[--var:var(--another)]\\"]",
  "[lg~=\\"bg-blue-600\\"]",
  "absolute",
  "fixed",
  "[important~=\\"text-red\\"]",
  "[important~=\\"bg-red\\"]",
  "[bg~=\\"blue-400\\"]",
  "[bg~=\\"hover:blue-500\\"]",
  "[bg~=\\"dark:!blue-500\\"]",
  "[bg~=\\"dark:hover:blue-600\\"]",
  "[text~=\\"sm\\"]",
  "[text~=\\"white\\"]",
  "[!leading-4=\\"\\"]",
  "[!leading-4=\\"true\\"]",
  "[flex~=\\"!~\\"]",
  "[flex~=\\"col\\"]",
  "[p~=\\"t-2\\"]",
  "[pt~=\\"2\\"]",
  "[border~=\\"rounded-xl\\"]",
  "[border~=\\"x-1\\"]",
  "[border~=\\"x-style-dashed\\"]",
  "[font~=\\"foo\\"]",
  "[font~=\\">\\"]",
  "[font~=\\"bar\\"]",
  "[font~=\\"?\\"]",
  "[font~=\\"mono\\"]",
  "[font~=\\":\\"]",
  "[font~=\\"sans\\"]",
  "[p~=\\"y-2\\"]",
  "[p~=\\"x-4\\"]",
  "[border~=\\"2\\"]",
  "[border~=\\"rounded\\"]",
  "[border~=\\"blue-200\\"]",
  "[un-children~=\\"m-auto\\"]",
  "[pt2=\\"\\"]",
  "[pt2=\\"true\\"]",
  "[rounded-sm=\\"\\"]",
  "[rounded-sm=\\"true\\"]",
  "[inline-block=\\"\\"]",
  "[inline-block=\\"true\\"]",
  "[transform=\\"\\"]",
  "[transform=\\"true\\"]",
  "[translate-x-100%=\\"\\"]",
  "[translate-x-100%=\\"true\\"]",
  "[translate-y-=\\"\\"]",
  "[translate-y-=\\"true\\"]",
  "[%=\\"\\"]",
  "[%=\\"true\\"]",
  "[rotate-30=\\"\\"]",
  "[rotate-30=\\"true\\"]",
  "[after~=\\"content-unocss\\"]",
  "[rotate-60=\\"\\"]",
  "[rotate-60=\\"true\\"]",
  "[ma=\\"\\"]",
  "[ma=\\"true\\"]",
  "[m~=\\"1\\"]",
  "[m~=\\"2\\"]",
  "[m~=\\"3\\"]",
}
`;

exports[`attributify > extractor2 1`] = `
Set {
  "[h-80=\\"\\"]",
  "[h-80=\\"true\\"]",
  "[text-center=\\"\\"]",
  "[text-center=\\"true\\"]",
  "[flex=\\"\\"]",
  "[flex=\\"true\\"]",
  "[flex-col=\\"\\"]",
  "[flex-col=\\"true\\"]",
  "[align-center=\\"\\"]",
  "[align-center=\\"true\\"]",
  "[select-none=\\"\\"]",
  "[select-none=\\"true\\"]",
  "[all:transition-400=\\"\\"]",
  "[all:transition-400=\\"true\\"]",
  "[type~=\\"checkbox\\"]",
  "[peer=\\"\\"]",
  "[peer=\\"true\\"]",
  "[mt-a=\\"\\"]",
  "[mt-a=\\"true\\"]",
  "[mb-a=\\"\\"]",
  "[mb-a=\\"true\\"]",
  "[group=\\"\\"]",
  "[group=\\"true\\"]",
  "[peer-checked~=\\"text-4xl\\"]",
  "[font-100=\\"\\"]",
  "[font-100=\\"true\\"]",
  "[text-4xl=\\"\\"]",
  "[text-4xl=\\"true\\"]",
  "[mb--3=\\"\\"]",
  "[mb--3=\\"true\\"]",
  "[p-10=\\"\\"]",
  "[p-10=\\"true\\"]",
  "[bg-gradient~=\\"to-r\\"]",
  "[bg-gradient~=\\"from-yellow-400\\"]",
  "[bg-gradient~=\\"via-red-500\\"]",
  "[bg-gradient~=\\"to-pink-500\\"]",
  "[text-5xl=\\"\\"]",
  "[text-5xl=\\"true\\"]",
  "[sm~=\\"bg-blue-600\\"]",
  "[op-20=\\"\\"]",
  "[op-20=\\"true\\"]",
  "[font-200=\\"\\"]",
  "[font-200=\\"true\\"]",
  "[mt-1=\\"\\"]",
  "[mt-1=\\"true\\"]",
  "[tracking-wider=\\"\\"]",
  "[tracking-wider=\\"true\\"]",
  "[group-hover~=\\"text-teal-400\\"]",
  "[group-hover~=\\"op-50\\"]",
  "[ma=\\"\\"]",
  "[ma=\\"true\\"]",
  "[inline-flex=\\"\\"]",
  "[inline-flex=\\"true\\"]",
  "[relative=\\"\\"]",
  "[relative=\\"true\\"]",
  "[h-14=\\"\\"]",
  "[h-14=\\"true\\"]",
  "[type~=\\"text\\"]",
  "[m-0=\\"\\"]",
  "[m-0=\\"true\\"]",
  "[pt-4=\\"\\"]",
  "[pt-4=\\"true\\"]",
  "[px-4=\\"\\"]",
  "[px-4=\\"true\\"]",
  "[text-true-gray-800=\\"\\"]",
  "[text-true-gray-800=\\"true\\"]",
  "[placeholder=\\"\\"]",
  "[placeholder=\\"true\\"]",
  "[absolute=\\"\\"]",
  "[absolute=\\"true\\"]",
  "[leading-1rem=\\"\\"]",
  "[leading-1rem=\\"true\\"]",
  "[left-4=\\"\\"]",
  "[left-4=\\"true\\"]",
  "[pointer-events-none=\\"\\"]",
  "[pointer-events-none=\\"true\\"]",
  "[text-gray-7=\\"\\"]",
  "[text-gray-7=\\"true\\"]",
  "[top~=\\"1/3\\"]",
  "[transition~=\\"200\\"]",
  "[transition~=\\"linear\\"]",
  "[peer-not-placeholder-shown~=\\"-translate-y-4\\"]",
  "[peer-not-placeholder-shown~=\\"scale-75\\"]",
  "[peer-not-placeholder-shown~=\\"origin-top-left\\"]",
  "[peer-not-placeholder-shown~=\\"text-green-500\\"]",
  "[peer-focus~=\\"-translate-y-4\\"]",
  "[peer-focus~=\\"scale-75\\"]",
  "[peer-focus~=\\"origin-top-left\\"]",
  "[peer-focus~=\\"text-green-500\\"]",
  "[before~=\\"content-!\\"]",
  "[after~=\\"content-[!]\\"]",
}
`;

exports[`attributify > fixture1 1`] = `
"/* layer: base */
[layer-base~=\\"c-white\\\\/10\\"]{color:rgba(255,255,255,0.1);}
[layer-base~=\\"hover\\\\:c-black\\\\/20\\"]:hover{color:rgba(0,0,0,0.2);}
/* layer: default */
.absolute{position:absolute;}
.fixed{position:fixed;}
[m~=\\"\\\\31 \\"]{margin:0.25rem;}
[m~=\\"\\\\32 \\"]{margin:0.5rem;}
[m~=\\"\\\\33 \\"]{margin:0.75rem;}
[ma=\\"\\"],
[ma=\\"true\\"],
[un-children~=\\"m-auto\\"]>*{margin:auto;}
[inline-block=\\"\\"],
[inline-block=\\"true\\"]{display:inline-block;}
[flex~=\\"\\\\!\\\\~\\"]{display:flex !important;}
[flex~=\\"col\\"]{flex-direction:column;}
[translate-x-100\\\\%=\\"\\"],
[translate-x-100\\\\%=\\"true\\"],
[rotate-30=\\"\\"],
[rotate-30=\\"true\\"],
[rotate-60=\\"\\"],
[rotate-60=\\"true\\"],
[transform=\\"\\"],
[transform=\\"true\\"]{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-transform:translateX(var(--un-translate-x)) translateY(var(--un-translate-y)) translateZ(var(--un-translate-z)) rotate(var(--un-rotate)) rotateX(var(--un-rotate-x)) rotateY(var(--un-rotate-y)) rotateZ(var(--un-rotate-z)) skewX(var(--un-skew-x)) skewY(var(--un-skew-y)) scaleX(var(--un-scale-x)) scaleY(var(--un-scale-y)) scaleZ(var(--un-scale-z));}
[translate-x-100\\\\%=\\"\\"],
[translate-x-100\\\\%=\\"true\\"]{--un-translate-x:100%;transform:var(--un-transform);}
[rotate-30=\\"\\"],
[rotate-30=\\"true\\"]{--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-rotate:30deg;transform:var(--un-transform);}
[rotate-60=\\"\\"],
[rotate-60=\\"true\\"]{--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-rotate:60deg;transform:var(--un-transform);}
[transform=\\"\\"],
[transform=\\"true\\"]{transform:var(--un-transform);}
[border~=\\"\\\\32 \\"]{border-width:2px;border-style:solid;}
[border~=\\"x-1\\"]{border-left-width:1px;border-right-width:1px;border-left-style:solid;border-right-style:solid;}
[border~=\\"blue-200\\"]{--un-border-opacity:1;border-color:rgba(191,219,254,var(--un-border-opacity));}
[border~=\\"rounded-xl\\"]{border-radius:0.75rem;}
[border~=\\"rounded\\"]{border-radius:0.25rem;}
[rounded-sm=\\"\\"],
[rounded-sm=\\"true\\"]{border-radius:0.125rem;}
[border~=\\"x-style-dashed\\"]{border-left-style:dashed;border-right-style:dashed;}
.dark [bg~=\\"dark\\\\:\\\\!blue-500\\"]{--un-bg-opacity:1 !important;background-color:rgba(59,130,246,var(--un-bg-opacity)) !important;}
.dark [bg~=\\"dark\\\\:hover\\\\:blue-600\\"]:hover{--un-bg-opacity:1;background-color:rgba(37,99,235,var(--un-bg-opacity));}
[bg~=\\"blue-400\\"]{--un-bg-opacity:1;background-color:rgba(96,165,250,var(--un-bg-opacity));}
[bg~=\\"hover\\\\:blue-500\\"]:hover{--un-bg-opacity:1;background-color:rgba(59,130,246,var(--un-bg-opacity));}
[important~=\\"bg-red\\"]{--un-bg-opacity:1 !important;background-color:rgba(248,113,113,var(--un-bg-opacity)) !important;}
[p~=\\"x-4\\"]{padding-left:1rem;padding-right:1rem;}
[p~=\\"y-2\\"]{padding-top:0.5rem;padding-bottom:0.5rem;}
[p~=\\"t-2\\"],
[pt~=\\"\\\\32 \\"],
[pt2=\\"\\"],
[pt2=\\"true\\"]{padding-top:0.5rem;}
[font~=\\"mono\\"]{font-family:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,\\"Liberation Mono\\",\\"Courier New\\",monospace;}
[font~=\\"sans\\"]{font-family:ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,\\"Segoe UI\\",Roboto,\\"Helvetica Neue\\",Arial,\\"Noto Sans\\",sans-serif,\\"Apple Color Emoji\\",\\"Segoe UI Emoji\\",\\"Segoe UI Symbol\\",\\"Noto Color Emoji\\";}
[text~=\\"sm\\"]{font-size:0.875rem;line-height:1.25rem;}
[\\\\!leading-4=\\"\\"],
[\\\\!leading-4=\\"true\\"]{line-height:1rem !important;}
[important~=\\"text-red\\"]{--un-text-opacity:1 !important;color:rgba(248,113,113,var(--un-text-opacity)) !important;}
[text~=\\"white\\"]{--un-text-opacity:1;color:rgba(255,255,255,var(--un-text-opacity));}
[after~=\\"content-unocss\\"]::after{content:\\"unocss\\";}
@media (min-width: 640px){
[sm~=\\"\\\\[color\\\\:red\\\\]\\"]{color:red;}
}
@media (min-width: 768px){
[md~=\\"\\\\[--var\\\\:var\\\\(--another\\\\)\\\\]\\"]{--var:var(--another);}
}
@media (min-width: 1024px){
[lg~=\\"bg-blue-600\\"]{--un-bg-opacity:1;background-color:rgba(37,99,235,var(--un-bg-opacity));}
}"
`;

exports[`attributify > fixture2 1`] = `
"/* layer: default */
[pointer-events-none=\\"\\"],
[pointer-events-none=\\"true\\"]{pointer-events:none;}
[absolute=\\"\\"],
[absolute=\\"true\\"]{position:absolute;}
[relative=\\"\\"],
[relative=\\"true\\"]{position:relative;}
[left-4=\\"\\"],
[left-4=\\"true\\"]{left:1rem;}
[top~=\\"\\\\31 \\\\/3\\"]{top:33.3333333333%;}
[m-0=\\"\\"],
[m-0=\\"true\\"]{margin:0rem;}
[ma=\\"\\"],
[ma=\\"true\\"]{margin:auto;}
[mb--3=\\"\\"],
[mb--3=\\"true\\"]{margin-bottom:-0.75rem;}
[mb-a=\\"\\"],
[mb-a=\\"true\\"]{margin-bottom:auto;}
[mt-1=\\"\\"],
[mt-1=\\"true\\"]{margin-top:0.25rem;}
[mt-a=\\"\\"],
[mt-a=\\"true\\"]{margin-top:auto;}
[h-14=\\"\\"],
[h-14=\\"true\\"]{height:3.5rem;}
[h-80=\\"\\"],
[h-80=\\"true\\"]{height:20rem;}
[flex=\\"\\"],
[flex=\\"true\\"]{display:flex;}
[inline-flex=\\"\\"],
[inline-flex=\\"true\\"]{display:inline-flex;}
[flex-col=\\"\\"],
[flex-col=\\"true\\"]{flex-direction:column;}
[peer=\\"\\"]:focus~[peer-focus~=\\"origin-top-left\\"],
[peer=\\"\\"]:not(:placeholder-shown)~[peer-not-placeholder-shown~=\\"origin-top-left\\"]{transform-origin:top left;}
[peer=\\"\\"]:focus~[peer-focus~=\\"-translate-y-4\\"],
[peer=\\"\\"]:not(:placeholder-shown)~[peer-not-placeholder-shown~=\\"-translate-y-4\\"],
[peer=\\"\\"]:focus~[peer-focus~=\\"scale-75\\"],
[peer=\\"\\"]:not(:placeholder-shown)~[peer-not-placeholder-shown~=\\"scale-75\\"]{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-transform:translateX(var(--un-translate-x)) translateY(var(--un-translate-y)) translateZ(var(--un-translate-z)) rotate(var(--un-rotate)) rotateX(var(--un-rotate-x)) rotateY(var(--un-rotate-y)) rotateZ(var(--un-rotate-z)) skewX(var(--un-skew-x)) skewY(var(--un-skew-y)) scaleX(var(--un-scale-x)) scaleY(var(--un-scale-y)) scaleZ(var(--un-scale-z));}
[peer=\\"\\"]:focus~[peer-focus~=\\"-translate-y-4\\"],
[peer=\\"\\"]:not(:placeholder-shown)~[peer-not-placeholder-shown~=\\"-translate-y-4\\"]{--un-translate-y:-1rem;transform:var(--un-transform);}
[peer=\\"\\"]:focus~[peer-focus~=\\"scale-75\\"],
[peer=\\"\\"]:not(:placeholder-shown)~[peer-not-placeholder-shown~=\\"scale-75\\"]{--un-scale-x:0.75;--un-scale-y:0.75;transform:var(--un-transform);}
[select-none=\\"\\"],
[select-none=\\"true\\"]{user-select:none;}
[bg-gradient~=\\"from-yellow-400\\"]{--un-gradient-from:rgba(250,204,21,var(--un-from-opacity, 1));--un-gradient-stops:var(--un-gradient-from), var(--un-gradient-to, rgba(255, 255, 255, 0));}
[bg-gradient~=\\"to-pink-500\\"]{--un-gradient-to:rgba(236,72,153,var(--un-to-opacity, 1));}
[bg-gradient~=\\"via-red-500\\"]{--un-gradient-stops:var(--un-gradient-from), rgba(239,68,68,var(--un-via-opacity, 1)), var(--un-gradient-to, rgba(255, 255, 255, 0));}
[bg-gradient~=\\"to-r\\"]{--un-gradient-shape:to right;--un-gradient:var(--un-gradient-shape), var(--un-gradient-stops);background-image:linear-gradient(var(--un-gradient));}
[p-10=\\"\\"],
[p-10=\\"true\\"]{padding:2.5rem;}
[px-4=\\"\\"],
[px-4=\\"true\\"]{padding-left:1rem;padding-right:1rem;}
[pt-4=\\"\\"],
[pt-4=\\"true\\"]{padding-top:1rem;}
[text-center=\\"\\"],
[text-center=\\"true\\"]{text-align:center;}
[peer=\\"\\"]:checked~[peer-checked~=\\"text-4xl\\"],
[text-4xl=\\"\\"],
[text-4xl=\\"true\\"]{font-size:2.25rem;line-height:2.5rem;}
[text-5xl=\\"\\"],
[text-5xl=\\"true\\"]{font-size:3rem;line-height:1;}
[font-100=\\"\\"],
[font-100=\\"true\\"]{font-weight:100;}
[font-200=\\"\\"],
[font-200=\\"true\\"]{font-weight:200;}
[leading-1rem=\\"\\"],
[leading-1rem=\\"true\\"]{line-height:1rem;}
[tracking-wider=\\"\\"],
[tracking-wider=\\"true\\"]{letter-spacing:0.05em;}
[group=\\"\\"]:hover [group-hover~=\\"text-teal-400\\"]{--un-text-opacity:1;color:rgba(45,212,191,var(--un-text-opacity));}
[peer=\\"\\"]:focus~[peer-focus~=\\"text-green-500\\"],
[peer=\\"\\"]:not(:placeholder-shown)~[peer-not-placeholder-shown~=\\"text-green-500\\"]{--un-text-opacity:1;color:rgba(34,197,94,var(--un-text-opacity));}
[text-gray-7=\\"\\"],
[text-gray-7=\\"true\\"]{--un-text-opacity:1;color:rgba(55,65,81,var(--un-text-opacity));}
[text-true-gray-800=\\"\\"],
[text-true-gray-800=\\"true\\"]{--un-text-opacity:1;color:rgba(38,38,38,var(--un-text-opacity));}
[group=\\"\\"]:hover [group-hover~=\\"op-50\\"]{opacity:0.5;}
[op-20=\\"\\"],
[op-20=\\"true\\"]{opacity:0.2;}
[all\\\\:transition-400=\\"\\"] *,
[all\\\\:transition-400=\\"true\\"] *{transition-property:color,background-color,border-color,text-decoration-color,fill,stroke,opacity,box-shadow,transform,filter,backdrop-filter;transition-timing-function:cubic-bezier(0.4, 0, 0.2, 1);transition-duration:400ms;}
[transition~=\\"\\\\32 00\\"]{transition-property:color,background-color,border-color,text-decoration-color,fill,stroke,opacity,box-shadow,transform,filter,backdrop-filter;transition-timing-function:cubic-bezier(0.4, 0, 0.2, 1);transition-duration:200ms;}
[after~=\\"content-\\\\[\\\\!\\\\]\\"]::after{content:!;}
[before~=\\"content-\\\\!\\"]::before{content:\\"!\\";}
@media (min-width: 640px){
[sm~=\\"bg-blue-600\\"]{--un-bg-opacity:1;background-color:rgba(37,99,235,var(--un-bg-opacity));}
}"
`;

exports[`attributify > variant 1`] = `
[
  "layer-base-c-white/10",
  "hover:layer-base-c-black/20",
  "sm-[color:red]",
  "md-[--var:var(--another)]",
  "lg-bg-blue-600",
  undefined,
  undefined,
  "important-text-red",
  "important-bg-red",
  "bg-blue-400",
  "hover:bg-blue-500",
  "dark:!bg-blue-500",
  "dark:hover:bg-blue-600",
  "text-sm",
  "text-white",
  "!leading-4",
  "!leading-4",
  "!flex",
  "flex-col",
  "p-t-2",
  "pt-2",
  "border-rounded-xl",
  "border-x-1",
  "border-x-style-dashed",
  "font-foo",
  "font->",
  "font-bar",
  "font-?",
  "font-mono",
  "font-:",
  "font-sans",
  "p-y-2",
  "p-x-4",
  "border-2",
  "border-rounded",
  "border-blue-200",
  "children-m-auto",
  "pt2",
  "pt2",
  "rounded-sm",
  "rounded-sm",
  "inline-block",
  "inline-block",
  "transform",
  "transform",
  "translate-x-100%",
  "translate-x-100%",
  "translate-y-",
  "translate-y-",
  "%",
  "%",
  "rotate-30",
  "rotate-30",
  "after-content-unocss",
  "rotate-60",
  "rotate-60",
  "ma",
  "ma",
  "m-1",
  "m-2",
  "m-3",
]
`;
